src/client.cpp

```diff
@@ -523,6 +523,62 @@ GAME_ERROR CGameLibRetro::GetRichPresenceEvaluation(char* evaluation, size_t siz
   return GAME_ERROR_NO_ERROR;
 }
 
+GAME_ERROR CGameLibRetro::ActivateAchievement(unsigned cheevo_id, const char* memaddr)
+{
+  CCheevos::Get().ActivateAchievement(cheevo_id, memaddr);
+  return GAME_ERROR_NO_ERROR;
+}
+
+
+GAME_ERROR CGameLibRetro::GetCheevo_ID(void (*Callback)(char* achievement_url, unsigned cheevo_id))
+{
+  if (!CCheevos::Get().GetCheevo_ID(Callback))
+  {
+    return GAME_ERROR_FAILED;
+  }
+  return GAME_ERROR_NO_ERROR();
+}
+
+GAME_ERROR CGameLibRetro::AwardAchievement(char* url,
+                                           size_t size,
+                                           const char* username,
+                                           const char* token,
+                                           unsigned cheevo_id,
+                                           int hardcore,
+                                           const char* game_hash)
+{
+  if (!CCheevos::Get().AwardAchievement(url, size, username, token, cheevo_id, hardcore, game_hash))
+  {
+    return GAME_ERROR_FAILED;
+  }
+  return GAME_ERROR_NO_ERROR;
+}
+
+/* GAME_ERROR CGameLibRetro::RuntimeEventHandler(char* url,
+                                              size_t size,
+                                              const char* username,
+                                              const char* token,
+                                              unsigned id,
+                                              unsigned cheevo_id,
+                                              int hardcore,
+                                              const char* game_hash)
+{
+  return GAME_ERROR_NO_ERROR;
+}
+*/
+GAME_ERROR CGameLibRetro::DeactivateTriggeredAchievement(unsigned cheevo_id)
+{
+  CCheevos::Get().DeactivateTriggeredAchievement(cheevo_id);
+  return GAME_ERROR_NO_ERROR;
```

Comment:
same here, return `GAME_ERROR_FAILED` if `DeactivateTriggeredAchievement` returns `false`

Reply:
I think there is no need of this function in client.cpp so I will remove it, as it is called from cheevos.cpp and used there only